### PR TITLE
Can now extract outputs of every SLURM processes

### PR DESCRIPTION
**Pull request by tristan0x**
Standard output and error of every processes spawned by SLURM
are redirected to dedicated files. It allows hpcbench to extract
metrics from everyone of those.

Therefore, structre of the `metrics` key in both YAML and JSON outputs
has changed.
Before it was a dictionary of metrics name -> value. Now it provides
metrics for every outputs of the command, for instance:

```yaml
metrics:
  -
    context:
      node: r1n1
      rank: 1
      executor: slurm
    measurement:
      latency: 1.1
  -
    context:
      node: r1n1
      rank: 2
      executor: slurm
    measurement:
      latency: 3
```

When commands are spawned with local executor, results look like:

```yaml
metrics:
  -
    context:
      executor: local
    measurement:
      latency: 1.1
```
Also:
* Remove plotting capabilities because unused
* Remove `metrics` key in benchmark YAML configuration because should
  have not been done in the first place.

### CODE REVIEW

File: hpcbench/api.py [hunk 1]
```diff
@@ -82,14 +83,24 @@ def metrics(self):
         """
         raise NotImplementedError  # pragma: no cover
 
+    @contextlib.contextmanager
+    def context(self, outdir, log_prefix):
```
Comment (ohm314) [inline hpcbench/api.py]: Since it's in api.py could you please add docstrings here?

File: hpcbench/driver.py [hunk 2]
```diff
@@ -696,14 +677,48 @@ def __call__(self, **kwargs):
...
+                                                 run_metrics)
+                    metrics.update(run_metrics)
+            rc = dict(context=log.context, measurement=metrics)
+            all_metrics.append(rc)
         return self.report
 
+    @property
+    def logs(self):
+
+        class LocalLog(namedtuple('LocalLog', ['path', 'log_prefix'])):
```
Comment (ohm314) [inline hpcbench/driver.py]: maybe a stupid question, but `logs` will be called plenty of times, every time re-creating `LocalLog` and `SrunLog` classes, doesn't that produce quite a bit of overhead that could be avoided by moving these two classes somewhere outside

File: tests/test_slurm.py [hunk 3]
```diff
@@ -32,7 +33,7 @@ def setUpClass(cls):
                 # output a job id
                 while [[ "$1" == -* ]] ; do shift ; done
                 export SLURMD_NODENAME={node}
-                source $@ > slurm-12345.out 2>&1
+                source $@
```
Comment (ohm314) [inline tests/test_slurm.py]: Why did you change this? It was mimicking the behavior of sbatch, producing a slrum-<jobid>.out file

File: tests/test_srun.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def setUpClass(cls):
                 # output a job id
                 while [[ "$1" == -* ]] ; do shift ; done
                 export SLURMD_NODENAME={node}
-                source $@ > slurm-12345.out 2>&1
+                source $@ >slurm-localhost-1.stdout 2>slurm-localhost-1.stderr
```
Comment (ohm314) [inline tests/test_srun.py]: dito

Comment (tristan0x) [reply tests/test_srun.py]: yes, and I should keep it that way.

File: hpcbench/driver.py (hunk 2 above)
Comment (tristan0x) [reply hpcbench/driver.py]: Overhead is minor, but good point though. I moved those 2 classes definition inside `MetricsDriver` class definition.

**Comment by tristan0x** [comment]
I took you remarks into account @ohm314. Let me know if you are ok. I will then squash commits.
Thank you
